tests/rados/test_scrub_omap.py

```diff
@@ -0,0 +1,119 @@
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test to create a large number of omap entries on the single PG pool and test osd resiliency
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    pool_obj = PoolFunctions(node=cephadm)
+
+    pool_target_configs = config["verify_osd_omap_entries"]["configurations"]
+    omap_target_configs = config["verify_osd_omap_entries"]["omap_config"]
+
+    # Creating pools and starting the test
+    for entry in pool_target_configs.values():
+        log.debug(
+            f"Creating {entry['pool_type']} pool on the cluster with name {entry['pool_name']}"
+        )
+        if entry.get("pool_type", "replicated") == "erasure":
+            method_should_succeed(
+                rados_obj.create_erasure_pool, name=entry["pool_name"], **entry
+            )
+        else:
+            method_should_succeed(
+                rados_obj.create_pool,
+                **entry,
+            )
+
+        log.debug(
+            "Created the pool. beginning to create large number of omap entries on the pool"
+        )
+    omap_thread = Thread(
+        target=create_omap, args=[pool_obj, pool_target_configs, omap_target_configs]
+    )
+
+    # verification of omap for 30 minutes
+    verification_osd_thread = Thread(target=verification_osd, args=[rados_obj, 30])
+    omap_thread.daemon = True
+    omap_thread.start()
+    verification_osd_thread.start()
+    verification_osd_thread.join()
+    omap_thread._delete()
+    if result == 0:
+        return 0
+    else:
+        return 1
+
+
+def create_omap(pool_fun, pool_configs, omap_configs):
+    """
+    Used to create a omap entries.
+    Args:
+        pool_fun : PoolFunctions object imported from  pool_workflows
+        pool_configs:  Pool configuration details
+        omap_target_configs: omap configurations
+
+    Returns: 1 if the method failed to create omap.
+
+    """
+
+    for entry in pool_configs.values():
+        if not pool_fun.fill_omap_entries(pool_name=entry["pool_name"], **omap_configs):
+            log.error(f"Omap entries not generated on pool {entry['pool_name']}")
+            return 1
+
+
+def verification_osd(rados_object, test_time):
+    """
+    This method is used to verify the OSD status while performing the scrub,deep-scrub and creating omaps
+    Args:
+        rados_object : RadosOrchestrator object
+        test_time: Used to set the test time
+
+    """
+    # The result parameter is used to return value in the thread
+    global result
+    result = 0
+    # performing scrub and deep-scrub
+    rados_object.run_scrub()
+    rados_object.run_deep_scrub()
```

Comment:
Instead of blind test time, Let's check for scrubbing to finish. I believe we have checks to test scrub start and end times.

Reply:
Checking after the scrub and deep-scrub.